[issue 105] - Adding the RHSSO Template provisioner

fabiobrz opened the pull request:
## Description
Fix #105 - ported to support testing of RH-SSO via templates, which is still being used by internal pipelines.

## Type of change
 - [ ] Bug fix (non-breaking change which fixes an issue)
 - [x] Feature (non-breaking change which adds functionality)
 - [ ] Breaking change (fix or feature that would cause existing functionality to change
 - [x] Chore (non-breaking change which doesn't affect codebase;
   test, version modification, documentation, etc.)

## Checklist
 - [x] Code contributed by me aligns with current project license: [Apache 2.0](https://www.apache.org/licenses/LICENSE-2.0)
 - [x] Pull Request contains a description of the changes
 - [x] Pull Request does not include fixes for multiple issues/topics
 - [x] Code is self-descriptive and/or documented
 - [x] I have implemented unit tests to cover my changes
 - [x] I tested my code in OpenShift

<!--
Integration tests (https://github.com/Intersmash/intersmash/tree/main/testsuite)
Please check integration tests and provide/improve tests if applicable.

Open your PR in Draft mode and verify all the applicable Checklist items before marking your pull request as ready for review
-->

fabiobrz commented:
/test all

fabiobrz commented:
/test all

fabiobrz commented:
/test all

fabiobrz commented:
/test prod-intersmash-e2e-prod

tommaso-borgato commented on tools/intersmash-tools-provisioners/src/main/java/org/jboss/intersmash/tools/application/openshift/RhSsoTemplateOpenShiftApplication.java:
```diff
@@ -0,0 +1,63 @@
+	 * @return Instance of {@link Path} representing a YAML definition for the desired realm configuration
+	 */
+	default Path getRealmConfigurationFilePath() {
+		return null;
```
@fabiobrz importing a realm isn't needed, right?

fabiobrz replied:
It depends, there are internal tests relying on this specific provisioner, hence I am going to check and get beck here, thanks for now!

fabiobrz replied:
Confirmed, it is used by existing EAP7 + RHSSO tests.

tommaso-borgato commented on tools/intersmash-tools-provisioners/src/main/java/org/jboss/intersmash/tools/util/keycloak/KeycloakAdminClient.java:
```diff
@@ -62,6 +62,7 @@ public KeycloakAdminClient(final String url, final String realm, final String us
 	public void importRealmConfiguration(InputStream is) throws IOException {
 		PartialImportRepresentation piRep = JsonSerialization.readValue(is, PartialImportRepresentation.class);
+		// TODO - fix, the realm name should be passed in as a parameter
 		keycloak.realm("wildfly-realm").partialImport(piRep);
```
@fabiobrz isn't the realm name specified in the JSON file to import?

fabiobrz replied:
Now that you've mentioned it, I see I have been maybe a bit too quick in here.
There's actually some controversial issue around the Keycloak/RH-SSO admin client (also about the compatible versions), and I am going to check it, thanks for bringing it up.

fabiobrz replied:
Back with some updates. So the answer is "no", meaning that the Keycloak Admin API contracts require for the realm name, to identify the realm where resources will be created.
Here's an example from an existing test:

```
{
  "roles": {
    "realm" : [{
      "name": "user"
    }]
  },

  "users": [{
    "username" : "user",
    "enabled": true,
    "credentials" : [{
      "type" : "password",
      "value" : "password"
    }],
    "realmRoles": ["user"],
    "clientRoles": {
      "account": ["view-profile", "manage-account"]
    }
  }]
}
```
Anyway, I fixed the method as well, according to the `TODO`. Thanks again for bringing this up.

tommaso-borgato commented:
@fabiobrz LGTM, I just left  couple of questions...

fabiobrz commented:
> @fabiobrz LGTM, I just left couple of questions...

Thanks @tommaso-borgato - I addressed your concerns, feel free to let me know in case you think something else is needed here.
